### DISCUSSION TITLE
Connecting to a different language server when one is already started.

### DISCUSSION THREAD

**Discussion by jgarte** [joaotavora/eglot#991] [Q&A]
When running eglot after a language server has already been started it doesn't let you choose a separate language server but just tries to reconnect to the current one. 

For example, I run `eglot` then I connect to `pylsp`.

I run `eglot` again and It asks me if I'd like to reconnect to the existing server instead of prompting to connect to a separate server like `pyright`.

Is this by design?

**Comment by joaotavora** [comment]
> Is this by design?

Yes.  Needing a reconnect is a very common thing. Some servers need it to refresh their view on a project.

You can't have more than one server connected at any given time for a given project/major-mode combination.

If you want to connect to another one, use `M-x eglot-disconnect` first.

If you find yourself connecting to `pylsp` by mistake instead of `pyright` as you wanted, then tweak `eglot-server-programs` so that mistake won't be repeated.

**Reply by jgarte** [reply]
@joaotavora By tweaking you mean put the preferred language server at the front of the list operated on by `eglot-alternatives` in the `eglot-server-programs` block?
